### PR TITLE
NoTicket - structured handling messages/connect/disconect

### PR DESCRIPTION
**Pull request by arvurb**

### CODE REVIEW

File: services/api/src/domain/messages.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    FigureMoved = 'FigureMoved',
+    NewBoardState = 'NewBoardState',
+}
+
+export interface Message {
+    type: MessageType,
+    payload: unknown,
+}
+
+export interface newPlayerMessage {
```
Comment (raywiis) [inline services/api/src/domain/messages.ts:16]: unused?

File: services/api/src/index.ts [hunk 2]
```diff
@@ -1,52 +1,29 @@
...
       ws.on('close', () => {
-        players.delete(newUser);
-        const allPlayers = Array.from(players.keys());
-        for (const connection of players.values()) {
-          if (connection.readyState === 1) {
-            connection.send(JSON.stringify(allPlayers), { binary: isBinary });
-          }
-        }
+        playersService.unsubscribe(ws);
+        playersService.playerDisconnected();
```
Comment (raywiis) [inline services/api/src/index.ts:25]: Why not a single function call?

File: services/api/src/index.ts [hunk 3]
```diff
@@ -1,52 +1,29 @@
...
   wss.handleUpgrade(req, socket, head, (ws, request) => {
     wss.emit('connection', ws, request);
-
-    ws.once('message', function message(data, isBinary) {
-      const newUser: user = {
-        id: uuid(),
-        name: data.toString(),
-      };
-      players.set(newUser, ws);
+    ws.once('message', (data) => {
```
Comment (raywiis) [inline services/api/src/index.ts:21]: If all messages are to be handled here, then
```diff
- ws.once('message', (data) => {
+ ws.on('message', (data) => {
```

File: services/api/src/index.ts (hunk 2 above)
Comment (arvurb) [reply services/api/src/index.ts:25]: just separation of concerns. Unsubscribe may not necessarily mean notifying all subscribers.

**Review by raywiis — APPROVED** [review]

**Review by jvalatkevicius — APPROVED** [review]

**Comment by raywiis** [comment]
Please resolve the lockfile conflict and we can close this :eyes:

I solved mine by deleting all lockfiles, node_modules and rerunning bootstrap on project root.

**Review by raywiis — APPROVED** [review]

**Pull request merged**